Mask test failures on power pc CPU.

There's a new travis build bot linux-ppc64le which uncovers some issues in the mask module.

See https://travis-ci.org/pygame/pygame/jobs/581695143

```
FAIL: test_from_surface__different_alphas_16bit (pygame.tests.mask_test.MaskModuleTest)

Ensures from_surface creates a mask with the correct bits set

----------------------------------------------------------------------

Traceback (most recent call last):

  File ".../pygame/tests/mask_test.py", line 4659, in test_from_surface__different_alphas_16bit

    self.assertEqual(mask.count(), expected_count, msg)

AssertionError: 240 != 176 : threshold=0

======================================================================

FAIL: test_from_surface__different_alphas_32bit (pygame.tests.mask_test.MaskModuleTest)

Ensures from_surface creates a mask with the correct bits set

----------------------------------------------------------------------

Traceback (most recent call last):

  File ".../pygame/tests/mask_test.py", line 4576, in test_from_surface__different_alphas_32bit

    self.assertEqual(mask.count(), expected_count, msg)

AssertionError: 255 != 191 : threshold=0

======================================================================

FAIL: test_overlap_mask__offset (pygame.tests.mask_test.MaskTypeTest)

Ensure an offset overlap_mask's mask is correctly calculated.

----------------------------------------------------------------------

Traceback (most recent call last):

  File ".../pygame/tests/mask_test.py", line 846, in test_overlap_mask__offset

    assertMaskEqual(self, overlap_mask, expected_mask, msg)

  File ".../pygame/tests/mask_test.py", line 109, in assertMaskEqual

    testcase.assertEqual(m1_count, m2.count(), msg=msg)

AssertionError: 128 != 0 : offset=(1, 1)
```

--- a/src_c/bitmask.c
+++ b/src_c/bitmask.c
@@ -82,6 +82,23 @@ bitcount(BITMASK_W n)
     }
 }
 
+/* Positive modulo of the given dividend and BITMASK_W_LEN as the divisor
+ * (dividend % BITMASK_W_LEN).
+ *
+ * Params:
+ *     dividend: dividend of the modulo operation, can be positive or negative
+ *
+ * Returns:
+ *     the positive modulo of: dividend % BITMASK_W_LEN, where the result will
+ *         be: 0 <= result < BITMASK_W_LEN
+ */
+static INLINE int
+bitmask_positive_modulo(int dividend)
+{
+    int result = dividend % BITMASK_W_LEN;
+    return (0 <= result) ? result : result + BITMASK_W_LEN;
+}
+
 bitmask_t *
 bitmask_create(int w, int h)
 {
@@ -171,9 +188,10 @@ bitmask_fill(bitmask_t *m)
 
     len = m->h * ((m->w - 1) / BITMASK_W_LEN);
 
-    shift = BITMASK_W_LEN - (m->w % BITMASK_W_LEN);
+    shift = bitmask_positive_modulo(BITMASK_W_LEN - m->w);
     full = ~(BITMASK_W)0;
     cmask = (~(BITMASK_W)0) >> shift;
+
     /* fill all the pixels that aren't in the rightmost BITMASK_Ws */
     for (pixels = m->bits; pixels < (m->bits + len); pixels++) {
         *pixels = full;
@@ -197,8 +215,9 @@ bitmask_invert(bitmask_t *m)
 
     len = m->h * ((m->w - 1) / BITMASK_W_LEN);
 
-    shift = BITMASK_W_LEN - (m->w % BITMASK_W_LEN);
+    shift = bitmask_positive_modulo(BITMASK_W_LEN - m->w);
     cmask = (~(BITMASK_W)0) >> shift;
+
     /* flip all the pixels that aren't in the rightmost BITMASK_Ws */
     for (pixels = m->bits; pixels < (m->bits + len); pixels++) {
         *pixels = ~(*pixels);
@@ -706,9 +725,11 @@ bitmask_overlap_mask(const bitmask_t *a, const bitmask_t *b, bitmask_t *c,
     if (xoffset + b->w > c->w) {
         BITMASK_W edgemask;
         int n = c->w / BITMASK_W_LEN;
-        shift = (n + 1) * BITMASK_W_LEN - c->w;
+
+        shift = bitmask_positive_modulo(BITMASK_W_LEN - c->w);
         edgemask = (~(BITMASK_W)0) >> shift;
         c_end = c->bits + n * c->h + MIN(c->h, b->h + yoffset);
+
         for (cp = c->bits + n * c->h + MAX(yoffset, 0); cp < c_end; cp++)
             *cp &= edgemask;
     }
@@ -852,9 +873,11 @@ bitmask_draw(bitmask_t *a, const bitmask_t *b, int xoffset, int yoffset)
     if (xoffset + b->w > a->w) {
         BITMASK_W edgemask;
         int n = a->w / BITMASK_W_LEN;
-        shift = (n + 1) * BITMASK_W_LEN - a->w;
+
+        shift = bitmask_positive_modulo(BITMASK_W_LEN - a->w);
         edgemask = (~(BITMASK_W)0) >> shift;
         a_end = a->bits + n * a->h + MIN(a->h, b->h + yoffset);
+
         for (ap = a->bits + n * a->h + MAX(yoffset, 0); ap < a_end; ap++)
             *ap &= edgemask;
     }

